### PR TITLE
Effect API

### PR DESCRIPTION
**Pull request by Ficus-x**

### CODE REVIEW

File: Exiled.API/Features/Effect.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        /// Gets or sets a value indicating whether the effect will add duration onto the effect if already active or not.
+        /// </summary>
+        [Description("If the effect is already active, setting to true will add this duration onto the effect.")]
+        public bool AddDurationIfActive { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the effect should be enabled or not.
+        /// </summary>
+        [Description("Indicates whether the effect should be enabled or not")]
+        public bool Enable { get; set; }
```
Comment (iopietro) [inline Exiled.API/Features/Effect.cs:71]:
```suggestion
        public bool IsEnabled { get; set; }
```

File: Exiled.API/Features/Player.cs [hunk 2]
```diff
@@ -2726,6 +2741,18 @@ public void EnableEffects(IEnumerable<EffectType> types, float duration = 0f, bo
...
+        /// <summary>
+        /// Enables a <see cref="IEnumerable{T}"/> of <see cref="Effect"/> on the player.
+        /// </summary>
+        /// <param name="effects">The <see cref="IEnumerable{T}"/> of <see cref="Effect"/> to enable.</param>
+        public void EnableEffects(IEnumerable<Effect> effects)
+        {
+            foreach (Effect effect in effects)
+            {
+                EnableEffect(effect);
+            }
```
Comment (iopietro) [inline Exiled.API/Features/Player.cs:2753]:
```suggestion
            foreach (Effect effect in effects)
                EnableEffect(effect);
```

File: Exiled.API/Features/Player.cs [hunk 3]
```diff
@@ -2693,6 +2693,21 @@ public void EnableEffect(EffectType type, float duration = 0f, bool addDurationI
                 EnableEffect(statusEffect, duration, addDurationIfActive);
         }
 
+        /// <summary>
+        /// Enables a <see cref="Effect">status effect</see> on the player.
+        /// </summary>
+        /// <param name="Effect">The <see cref="Effect"/> to enable.</param>
+        private void EnableEffect(Effect effect)
+        {
+            if (effect.Enable)
```
Comment (iopietro) [inline Exiled.API/Features/Player.cs:2702]:
```suggestion
            if (effect.IsEnabled)
```

File: Exiled.API/Features/Effect.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        /// <param name="addDurationIfActive">Whether the effect will add duration onto the effect if already active or not.</param>
+        /// <param name="enable">Whether or not the effect should be enabled.</param>
+        public Effect(EffectType type, float duration, byte intensity = 1, bool addDurationIfActive = false, bool enable = true)
+        {
+            Type = type;
+            Duration = duration;
+            Intensity = intensity;
+            AddDurationIfActive = addDurationIfActive;
+            Enable = enable;
+        }
```
Comment (iopietro) [inline Exiled.API/Features/Effect.cs:41]:
```suggestion
        /// <param name="isEnabled">Whether or not the effect should be enabled.</param>
        public Effect(EffectType type, float duration, byte intensity = 1, bool addDurationIfActive = false, bool isEnabled = true)
        {
            Type = type;
            Duration = duration;
            Intensity = intensity;
            AddDurationIfActive = addDurationIfActive;
            IsEnabled = isEnabled;
        }
```

**Review by iopietro — CHANGES_REQUESTED** [review]

**Review by iopietro — APPROVED** [review]

File: Exiled.API/Features/Player.cs [hunk 5]
```diff
@@ -2693,6 +2693,21 @@ public void EnableEffect(EffectType type, float duration = 0f, bool addDurationI
                 EnableEffect(statusEffect, duration, addDurationIfActive);
         }
 
+        /// <summary>
+        /// Enables a <see cref="Effect">status effect</see> on the player.
+        /// </summary>
+        /// <param name="Effect">The <see cref="Effect"/> to enable.</param>
+        private void EnableEffect(Effect effect)
```
Comment (Thundermaker300) [inline Exiled.API/Features/Player.cs:2700]:
```suggestion
        public void EnableEffect(Effect effect)
```
🤨

Comment (Ficus-x) [reply Exiled.API/Features/Player.cs:2700]: LOL

File: Exiled.API/Features/Player.cs [hunk 6]
```diff
@@ -2694,6 +2694,21 @@ public void EnableEffect(EffectType type, float duration = 0f, bool addDurationI
                 EnableEffect(statusEffect, duration, addDurationIfActive);
         }
 
+        /// <summary>
+        /// Enables a <see cref="Effect">status effect</see> on the player.
+        /// </summary>
+        /// <param name="Effect">The <see cref="Effect"/> to enable.</param>
```
Comment (louis1706) [inline Exiled.API/Features/Player.cs:2700]: Fix build Error

**Review by louis1706 — CHANGES_REQUESTED** [review]

**Review by louis1706 — APPROVED** [review]

File: Exiled.API/Features/Player.cs [hunk 7]
```diff
@@ -2694,6 +2694,21 @@ public void EnableEffect(EffectType type, float duration = 0f, bool addDurationI
...
+        /// Enables a <see cref="Effect">status effect</see> on the player.
+        /// </summary>
+        /// <param name="effect">The <see cref="Effect"/> to enable.</param>
+        public void EnableEffect(Effect effect)
+        {
+            if (effect.IsEnabled)
+            {
+                EnableEffect(effect.Type, effect.Duration, effect.AddDurationIfActive);
+
+                if (effect.Intensity > 0)
```
Comment (louis1706) [inline Exiled.API/Features/Player.cs:2707]: this should be > 1 no ?

Comment (Ficus-x) [reply Exiled.API/Features/Player.cs:2707]: its > 0 for not errors because intensity begins with 1

**Review by iopietro — APPROVED** [review]

**Pull request merged**